library/core/src/mem/mod.rs

```diff
@@ -637,8 +636,6 @@ pub const fn needs_drop<T: ?Sized>() -> bool {
 #[inline(always)]
 #[must_use]
 #[stable(feature = "rust1", since = "1.0.0")]
-#[allow(deprecated_in_future)]
-#[allow(deprecated)]
```

Comment:
I noticed these unnecessary `allow(deprecated)` while grepping through the file, and removed them as a drive-by fix.